Changes

```diff
@@ -767,6 +767,9 @@ OCaml 4.12.0
 - #10048: Fix bug with generalized local opens.
   (Leo White, review by Thomas Refis)
 
+- #10106, #10112: preserve expected-type explanation after "let () = ... in"
```

Comment:
The fix is quite broader than just `let ( ) = ... in` . Maybe

> after some instances of `let ... = ... in` and `try ... with`

?

Reply:
Perhaps something like: Preserve the type error explanation in conditionals with a non-unit let-expression in the "if" branch and a missing "else" branch.

Reply:
I rephrased the Changes entry, thanks.